backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/NormService.java

```diff
@@ -32,7 +33,7 @@ public class NormService
   private final LoadNormPort loadNormPort;
   private final LoadNormByGuidPort loadNormByGuidPort;
   private final UpdateNormPort updateNormPort;
-  private final ModificationValidator modificationValidator;
+  private final ValidationService validationService;
```

Comment:
Could we not "just" use a `SingleModValidator` here and remove the need for both `ValidatorService` and `ValidatorName`? I don't see the advantage that the `ValidatorService` provides us with at the moment 🧐

Reply:
We could do that yes, but I think that encapsulating the logic of which validator to use can be deferred to the `ValidationService`. Later on we could also have the case that we need to chain several validators and this could be done by the `ValidationService`. Also we keep our backend package structure cleaner by having only services within the `service` package.

Reply:
We could yes but I think it is better if we deferred the logic of which validator to use to a service. Maybe later we want to chain validators and this could be done in the `ValidationService`.

Reply:
How would you add chaining logic here? I'm really unsure how that would simplify any any work by using the `ValidationService` instead of calling two (or more) Validators. But having the `ValidationService` definitely adds some complexity.

I'm also not sure I understand how this change affects the package structure

Reply:
I had an idea in mind with the chaining but I just realized it was using the `Validator` interface... so forget that wrong argumentation.

I created the `ValidationService` after having refactored the validators (meaning separating the two) and, at least in my head, they are not real services but rather smaller components doing just 1 thing: validating. So that is why I created the `ValidationService` to act as the "manager" of the validators.

Every future custom validator will implement the `Validator` interface but will not act as a service, or at least as the services we had till now (meaning call coming from the controller and using a use case which is implemented in a service). But well you could also say the `ValidationService` is not implementing a use case but we could introduce one or two. Also calling a validator a service feels wrong but it may just be my feeling.

But yeah this argumentation would not be a reason enough, now that I try to argument the idea, to having the `ValidationService`.  But maybe in the future we will extend what the ValidationService can do, like for example seting flags for throwing exceptions or do a dry-run. 

But please if you don't see any benefit or future application, keep challenging it and I would remove it.

Reply:
Oh forgot, the thing with the package structure is we had before only `port` and `service` and I introduced now the `validator` package, where we could put all validators annotated with @Component (I know is just the generic annotation)

Reply:
I don't really see why a validator shouldn't be a service, but thats not a hill i will die on ^^ So I'm totally fine with having them separated and using the `@component` annotation. (I actually see an advantage with having them in another folder to keep the service folder smaller)

I'm still not convinced that the benefits you see in the `ValidatorService` outweigh the problems it could create (also for type-safety as you mentioned in the other comment) but as we just discussed we can move all this to a later time :)